### DISCUSSION TITLE
About src ip address and dest ip address in tcp handshake

### DISCUSSION THREAD

**Discussion by 03psb** [ANLAB-KAIST/KENSv3#220] [Project Q&A 2. Transport Layer (TCP)]
I'm working on handshake testcase, particularly backlog1 one. In this testcase, packet is moved between address 192.168.0.7 and 10.0.1.4
<img width="479" alt="question" src="https://github.com/ANLAB-KAIST/KENSv3/assets/98043513/ce8adf49-316a-4dec-a0d1-3baa9f587303">
But as you can see on the picture that I read packet source address and destination address, it had no relavance with nor 192.168.0.7 abd 10.0.1.4. and it even seemed to change a bit whenever I re-run the testcase. Does some kind of network address translation going on inside KENS? even so, I don't suspect they are real addresses, since sending the packet to those are failed. Anyway, I want to know which is the case

**Comment by Pusnow** [comment]
> Does some kind of network address translation going on inside KENS?

No

It seems that you parsed packets incorrectly or misuse `inet_ntoa`. Please double check these.
